src/uct/ib/ud/base/ud_ep.c

```diff
@@ -526,12 +514,22 @@ uct_ud_send_skb_t *uct_ud_ep_prepare_creq(uct_ud_ep_t *ep)
 
     creq->type                    = UCT_UD_PACKET_CREQ;
     creq->conn_req.conn_id        = ep->conn_id;
-    memcpy(&creq->conn_req.ib_addr, &ib_addr, sizeof(ib_addr));
-    memcpy(&creq->conn_req.ep_addr, &ep_addr, sizeof(ep_addr));
+
+    status = uct_ud_ep_get_address(&ep->super.super,
+                                   (void*)&creq->conn_req.ep_addr);
+    if (status != UCS_OK) {
+        return NULL;
+    }
+
+    status = uct_ib_iface_get_device_address(&iface->super.super.super,
+                                             (uct_device_addr_t*)uct_ud_creq_ib_addr(creq));
+    if (status != UCS_OK) {
+        return NULL;
+    }
```

Comment:
i can't take address earlier since i have to provide buffer to pack the address into (which is the skb)..